pkg/operation/botanist/namespaces.go

```diff
@@ -146,6 +154,23 @@ func (b *Botanist) DeploySeedNamespace(ctx context.Context) error {
 	return nil
 }
 
+func extractZonesFromNodeSelectorTerm(term corev1.NodeSelectorTerm) []string {
+	zones := sets.NewString()
+	for _, matchExpression := range term.MatchExpressions {
+		key := matchExpression.Key
+		// Only consider labels with 'topology.{provider-specific-string}/zone' which should match most of the cases.
+		if (!strings.HasPrefix(key, "topology.") && !strings.HasSuffix(key, "/zone")) ||
+			key != corev1.LabelFailureDomainBetaZone {
+			continue
+		}
+		if matchExpression.Operator != corev1.NodeSelectorOpIn {
+			continue
+		}
+		zones.Insert(matchExpression.Values...)
+	}
+	return zones.UnsortedList()
```

Comment:
Q: Doesn't `UnsortedList` lead to undefined order of zones which could flake a test?